### ISSUE TITLE
Stuck at meshing

### ISSUE THREAD

**Issue by Xerxesious** [alicevision/meshroom#243]
Hey Guys

This is my log entry:

Program called with the following parameters:
 * angleFactor = 15
 * contributeMarginFactor = 2
 * depthMapFilterFolder = "C:/Users/Mat/Desktop/Mat/Tree/Cropped/MeshroomCache/DepthMapFilter/8fdb6e15a6aa7efd82a88332f8d75a0bb5239b8f"
 * depthMapFolder = "C:/Users/Mat/Desktop/Mat/Tree/Cropped/MeshroomCache/DepthMap/b529a5506adae74ac7bbda0db1071c4e4ff50256"
 * ini = "C:/Users/Mat/Desktop/Mat/Tree/Cropped/MeshroomCache/PrepareDenseScene/6c539683e60767f7e52bc5fc8ffe2bdd0bb2a1a7/mvs.ini"
 * maxInputPoints = 50000000
 * maxPoints = 5000000
 * maxPointsPerVoxel = 1000000
 * minAngleThreshold = 1
 * minStep = 2
 * output = "C:/Users/Mat/Desktop/Mat/Tree/Cropped/MeshroomCache/Meshing/8a4f9a21ab0b7832ba76bfb0a336ff59d893b57a/mesh.obj"
 * partitioning =  Unknown Type "enum EPartitioningMode"
 * pixSizeMarginFinalCoef = 4
 * pixSizeMarginInitCoef = 2
 * refineFuse = 1
 * repartition =  Unknown Type "enum ERepartitionMode"
 * simFactor = 15
 * simGaussianSize = 10
 * simGaussianSizeInit = 10
 * verboseLevel = "info"
 * voteMarginFactor = 4

[19:08:13.840709][info] Found 1 image dimension(s): 
[19:08:13.840709][info]  - [3024x3504]
[19:08:13.871958][info] Overall maximum dimension: [1512x1752]
[19:08:13.871958][warning] repartitionMode: 1
[19:08:13.871958][warning] partitioningMode: 1
[19:08:13.871958][info] Meshing mode: multi-resolution, partitioning: single block.
[19:08:13.871958][info] Estimate space.

I think my problem has something to do with the repartition and partitioning.  So I changed those in the attributes section and the program still did not proceed to MeshFiltering.

**Comment by Henri2h** [comment]
Hey Guys,
I have the same issue :


Program called with the following parameters:
 * angleFactor = 15
 * contributeMarginFactor = 2
 * depthMapFilterFolder = "C:/Users/henri/3D Objects/plateau 2/MeshroomCache/DepthMapFilter/9ac2bcd2ca8fdbf1b183ee1ea6938bdd10ee6701"
 * depthMapFolder = "C:/Users/henri/3D Objects/plateau 2/MeshroomCache/DepthMap/a51782665925232ed03086213a95d327f041a21e"
 * ini = "C:/Users/henri/3D Objects/plateau 2/MeshroomCache/PrepareDenseScene/37c3e70e216d9c3e9b0aec4fb3d4e27c4c239fbc/mvs.ini"
 * maxInputPoints = 55647000
 * maxPoints = 1330000
 * maxPointsPerVoxel = 1711000
 * minAngleThreshold = 1
 * minStep = 2
 * output = "C:/Users/henri/3D Objects/plateau 2/MeshroomCache/Meshing/1c53a445e6afdcb503a1be78472b1a1460a1ff02/mesh.obj"
 * partitioning =  Unknown Type "enum EPartitioningMode"
 * pixSizeMarginFinalCoef = 4
 * pixSizeMarginInitCoef = 2
 * refineFuse = 1
 * repartition =  Unknown Type "enum ERepartitionMode"
 * simFactor = 15
 * simGaussianSize = 10
 * simGaussianSizeInit = 10
 * verboseLevel = "info"
 * voteMarginFactor = 4

[22:31:56.074699][info] Found 1 image dimension(s): 
[22:31:56.080684][info]  - [2448x3264]
[22:31:56.124560][info] Overall maximum dimension: [1224x1632]
[22:31:56.125557][warning] repartitionMode: 1
[22:31:56.125557][warning] partitioningMode: 2

**Comment by pwab** [comment]
Same here

### HARDWARE
 * Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz
 * NVIDIA GeForce GTX 1050 Ti (GP107-400) [768 CUDA-cores]
 * 16GB Crucial DDR4-2400

### SOFTWARE
 * Windows 10 Pro 1803 64-bit
 * [Meshroom 2018.1.0 (2018/08/09)](https://github.com/alicevision/meshroom/commit/e5376b9c93eaed1eac32454064b3af327a691980)
 * NVIDIA Driver 399.07
 * NVIDIA CUDA 9.2.217

### LOG
```
[10/12] Meshing
 - commandLine: aliceVision_meshing  --ini "C:/Users/pwab/Desktop/schluessel/MeshroomCache/PrepareDenseScene/fe709afcee752880f360f8ef985f4e9f102887f8/mvs.ini" --depthMapFolder "C:/Users/pwab/Desktop/schluessel/MeshroomCache/DepthMap/7a545cb312fb203153d908913eb55d6fec28041d" --depthMapFilterFolder "C:/Users/pwab/Desktop/schluessel/MeshroomCache/DepthMapFilter/fb1cf62dd9d7d199fad8ea5fdaaf399433218a75" --maxInputPoints 50000000 --maxPoints 5000000 --maxPointsPerVoxel 1000000 --minStep 2 --partitioning singleBlock --repartition multiResolution --angleFactor 15.0 --simFactor 15.0 --pixSizeMarginInitCoef 2.0 --pixSizeMarginFinalCoef 4.0 --voteMarginFactor 4.0 --contributeMarginFactor 2.0 --simGaussianSizeInit 10.0 --simGaussianSize 10.0 --minAngleThreshold 1.0 --refineFuse True --verboseLevel trace --output "C:/Users/pwab/Desktop/schluessel/MeshroomCache/Meshing/1aaa527e9d806face18de45de175750bb368d50a/mesh.obj"
 - logFile: C:/Users/pwab/Desktop/schluessel/MeshroomCache/Meshing/1aaa527e9d806face18de45de175750bb368d50a/log
 - elapsed time: 0:00:01.801734
ERROR:root:Error on node computation: Error on node "Meshing_1":
Log:
Program called with the following parameters:
 * angleFactor = 15
 * contributeMarginFactor = 2
 * depthMapFilterFolder = "C:/Users/pwab/Desktop/schluessel/MeshroomCache/DepthMapFilter/fb1cf62dd9d7d199fad8ea5fdaaf399433218a75"
 * depthMapFolder = "C:/Users/pwab/Desktop/schluessel/MeshroomCache/DepthMap/7a545cb312fb203153d908913eb55d6fec28041d"
 * ini = "C:/Users/pwab/Desktop/schluessel/MeshroomCache/PrepareDenseScene/fe709afcee752880f360f8ef985f4e9f102887f8/mvs.ini"
 * maxInputPoints = 50000000
 * maxPoints = 5000000
 * maxPointsPerVoxel = 1000000
 * minAngleThreshold = 1
 * minStep = 2
 * output = "C:/Users/pwab/Desktop/schluessel/MeshroomCache/Meshing/1aaa527e9d806face18de45de175750bb368d50a/mesh.obj"
 * partitioning =  Unknown Type "enum EPartitioningMode"
 * pixSizeMarginFinalCoef = 4
 * pixSizeMarginInitCoef = 2
 * refineFuse = 1
 * repartition =  Unknown Type "enum ERepartitionMode"
 * simFactor = 15
 * simGaussianSize = 10
 * simGaussianSizeInit = 10
 * verboseLevel = "trace"
 * voteMarginFactor = 4

[10:57:20.980565][info] Found 2 image dimension(s):
[10:57:20.981539][info]  - [1944x2592]
[10:57:20.981539][info]  - [2592x1944]
[10:57:20.981539][debug] [IO] Read Image Metadata: C:/Users/pwab/Desktop/schluessel/MeshroomCache/DepthMap/7a545cb312fb203153d908913eb55d6fec28041d/812854499_depthMap.exr
[10:57:20.983532][debug] [IO] Read Image Metadata: C:/Users/pwab/Desktop/schluessel/MeshroomCache/DepthMap/7a545cb312fb203153d908913eb55d6fec28041d/1261943489_depthMap.exr
[10:57:20.984554][info] Overall maximum dimension: [1296x1296]
[10:57:20.984554][warning] repartitionMode: 1
[10:57:20.985551][warning] partitioningMode: 1
[10:57:20.985551][info] Meshing mode: multi-resolution, partitioning: single block.
[10:57:20.985551][info] Estimate space.
[10:57:20.986531][debug] [IO] Read Image Metadata: C:/Users/pwab/Desktop/schluessel/MeshroomCache/DepthMapFilter/fb1cf62dd9d7d199fad8ea5fdaaf399433218a75/812854499_depthMap.exr
[10:57:20.986531][debug] [IO] Read Image Metadata: C:/Users/pwab/Desktop/schluessel/MeshroomCache/DepthMapFilter/fb1cf62dd9d7d199fad8ea5fdaaf399433218a75/1261943489_depthMap.exr
[10:57:20.988519][debug] [IO] Read Image: C:/Users/pwab/Desktop/schluessel/MeshroomCache/DepthMapFilter/fb1cf62dd9d7d199fad8ea5fdaaf399433218a75/812854499_depthMap.exr
[10:57:21.012455][debug] [IO] Read Image: C:/Users/pwab/Desktop/schluessel/MeshroomCache/DepthMapFilter/fb1cf62dd9d7d199fad8ea5fdaaf399433218a75/1261943489_depthMap.exr

WARNING: downgrade status on node "MeshFiltering_1" from Status.SUBMITTED to Status.NONE
WARNING: downgrade status on node "Texturing_1" from Status.SUBMITTED to Status.NONE
ERROR:root:Error during Graph execution Error on node "Meshing_1":
Log:
Program called with the following parameters:
 * angleFactor = 15
 * contributeMarginFactor = 2
 * depthMapFilterFolder = "C:/Users/pwab/Desktop/schluessel/MeshroomCache/DepthMapFilter/fb1cf62dd9d7d199fad8ea5fdaaf399433218a75"
 * depthMapFolder = "C:/Users/pwab/Desktop/schluessel/MeshroomCache/DepthMap/7a545cb312fb203153d908913eb55d6fec28041d"
 * ini = "C:/Users/pwab/Desktop/schluessel/MeshroomCache/PrepareDenseScene/fe709afcee752880f360f8ef985f4e9f102887f8/mvs.ini"
 * maxInputPoints = 50000000
 * maxPoints = 5000000
 * maxPointsPerVoxel = 1000000
 * minAngleThreshold = 1
 * minStep = 2
 * output = "C:/Users/pwab/Desktop/schluessel/MeshroomCache/Meshing/1aaa527e9d806face18de45de175750bb368d50a/mesh.obj"
 * partitioning =  Unknown Type "enum EPartitioningMode"
 * pixSizeMarginFinalCoef = 4
 * pixSizeMarginInitCoef = 2
 * refineFuse = 1
 * repartition =  Unknown Type "enum ERepartitionMode"
 * simFactor = 15
 * simGaussianSize = 10
 * simGaussianSizeInit = 10
 * verboseLevel = "trace"
 * voteMarginFactor = 4

[10:57:20.980565][info] Found 2 image dimension(s):
[10:57:20.981539][info]  - [1944x2592]
[10:57:20.981539][info]  - [2592x1944]
[10:57:20.981539][debug] [IO] Read Image Metadata: C:/Users/pwab/Desktop/schluessel/MeshroomCache/DepthMap/7a545cb312fb203153d908913eb55d6fec28041d/812854499_depthMap.exr
[10:57:20.983532][debug] [IO] Read Image Metadata: C:/Users/pwab/Desktop/schluessel/MeshroomCache/DepthMap/7a545cb312fb203153d908913eb55d6fec28041d/1261943489_depthMap.exr
[10:57:20.984554][info] Overall maximum dimension: [1296x1296]
[10:57:20.984554][warning] repartitionMode: 1
[10:57:20.985551][warning] partitioningMode: 1
[10:57:20.985551][info] Meshing mode: multi-resolution, partitioning: single block.
[10:57:20.985551][info] Estimate space.
[10:57:20.986531][debug] [IO] Read Image Metadata: C:/Users/pwab/Desktop/schluessel/MeshroomCache/DepthMapFilter/fb1cf62dd9d7d199fad8ea5fdaaf399433218a75/812854499_depthMap.exr
[10:57:20.986531][debug] [IO] Read Image Metadata: C:/Users/pwab/Desktop/schluessel/MeshroomCache/DepthMapFilter/fb1cf62dd9d7d199fad8ea5fdaaf399433218a75/1261943489_depthMap.exr
[10:57:20.988519][debug] [IO] Read Image: C:/Users/pwab/Desktop/schluessel/MeshroomCache/DepthMapFilter/fb1cf62dd9d7d199fad8ea5fdaaf399433218a75/812854499_depthMap.exr
[10:57:21.012455][debug] [IO] Read Image: C:/Users/pwab/Desktop/schluessel/MeshroomCache/DepthMapFilter/fb1cf62dd9d7d199fad8ea5fdaaf399433218a75/1261943489_depthMap.exr
```

**Comment by chrisnagy** [comment]
First time for me today Stuck on Meshing Never had a problem b4 the model is a bit more complex wiht 214 images

**Comment by DevoStevo66** [comment]
I've also got stuck on Meshing node for this model. Accepted all images and the SFM in 3D view looks fine but won't finish meshing. I've had success with several other Meshroom models, just not sure why it hangs up on this one.  I did notice at the end of the log the "Overall maximum dimension:" is "1368 x 1368" whereas my image size is 2736 x 2736. Not sure if this has any relevance but did seem strange when mine are larger in size. Also, Meshroom has worked fine before using the same image size, camera and settings (unless I accidentally changed the default setting at some stage in Meshroom settings and saved without realizing it).

Here's my log file.

Plugins loaded:  CameraCalibration, CameraConnection, CameraInit, CameraLocalization, CameraRigCalibration, CameraRigLocalization, ConvertAnimatedCamera, ConvertSfMFormat, DepthMap, DepthMapFilter, ExportAnimatedCamera, ExportMaya, ExportUndistortedImages, FeatureExtraction, FeatureMatching, ImageMatching, ImageMatchingMultiSfM, KeyframeSelection, MeshDecimate, MeshDenoising, MeshFiltering, MeshResampling, Meshing, PrepareDenseScene, Publish, SfMAlignment, SfMTransform, StructureFromMotion, Texturing
[QtOIIO] Plugin Initialized
[DepthMapEntity] DepthMapEntity
Failed to download scene at QUrl("")
Nodes to execute:  ['Meshing_1']
WARNING: downgrade status on node "Meshing_1" from Status.ERROR to Status.SUBMITTED

[1/1] Meshing
 - commandLine: aliceVision_meshing  --ini "C:/Users/...../MeshroomCache/PrepareDenseScene/61c4f02547ec2c49ec920d480c2c166a0baa7318/mvs.ini" --depthMapFolder "C:/Users/Steven/Documents/3DProjects/3DTrees/Illawarra Flame trunk/MeshroomCache/DepthMap/9db3b9c42b7eaebdcb2515dd5f9482106b2040af" --depthMapFilterFolder "C:/Users/Steven/Documents/3DProjects/3DTrees/Illawarra Flame trunk/MeshroomCache/DepthMapFilter/de97a40f6dbb036e00ff4b0ff4a4ea7f24d63e1e" --maxInputPoints 50000000 --maxPoints 5000000 --maxPointsPerVoxel 1000000 --minStep 2 --partitioning auto --repartition multiResolution --angleFactor 15.0 --simFactor 15.0 --pixSizeMarginInitCoef 2.0 --pixSizeMarginFinalCoef 4.0 --voteMarginFactor 4.0 --contributeMarginFactor 2.0 --simGaussianSizeInit 10.0 --simGaussianSize 10.0 --minAngleThreshold 1.0 --refineFuse True --verboseLevel info --output "C:/Users/Steven/Documents/3DProjects/3DTrees/Illawarra Flame trunk/MeshroomCache/Meshing/79d6fdbd49f1152dbfd93532840c9392e179ff4e/mesh.obj"
 - logFile: C:/Users/....../MeshroomCache/Meshing/79d6fdbd49f1152dbfd93532840c9392e179ff4e/log
 - elapsed time: 0:00:04.679559
ERROR:root:Error on node computation: Error on node "Meshing_1":
Log:
Program called with the following parameters:
 * angleFactor = 15
 * contributeMarginFactor = 2
 * depthMapFilterFolder = "C:/Users/....../MeshroomCache/DepthMapFilter/de97a40f6dbb036e00ff4b0ff4a4ea7f24d63e1e"
 * depthMapFolder = "C:/Users/....../MeshroomCache/DepthMap/9db3b9c42b7eaebdcb2515dd5f9482106b2040af"
 * ini = "C:/Users/....../MeshroomCache/PrepareDenseScene/61c4f02547ec2c49ec920d480c2c166a0baa7318/mvs.ini"
 * maxInputPoints = 50000000
 * maxPoints = 5000000
 * maxPointsPerVoxel = 1000000
 * minAngleThreshold = 1
 * minStep = 2
 * output = "C:/Users/....../MeshroomCache/Meshing/79d6fdbd49f1152dbfd93532840c9392e179ff4e/mesh.obj"
 * partitioning =  Unknown Type "enum EPartitioningMode"
 * pixSizeMarginFinalCoef = 4
 * pixSizeMarginInitCoef = 2
 * refineFuse = 1
 * repartition =  Unknown Type "enum ERepartitionMode"
 * simFactor = 15
 * simGaussianSize = 10
 * simGaussianSizeInit = 10
 * verboseLevel = "info"
 * voteMarginFactor = 4

[15:37:09.160977][info] Found 1 image dimension(s):
[15:37:09.165975][info]  - [2736x2736]
[15:37:09.367901][info] Overall maximum dimension: [1368x1368]
[15:37:09.369897][warning] repartitionMode: 1
[15:37:09.370896][warning] partitioningMode: 2

ERROR:root:Error during Graph execution Error on node "Meshing_1":
Log:
Program called with the following parameters:
 * angleFactor = 15
 * contributeMarginFactor = 2
 * depthMapFilterFolder = "C:/Users/....../MeshroomCache/DepthMapFilter/de97a40f6dbb036e00ff4b0ff4a4ea7f24d63e1e"
 * depthMapFolder = "C:/Users/...../MeshroomCache/DepthMap/9db3b9c42b7eaebdcb2515dd5f9482106b2040af"
 * ini = "C:/Users/....../MeshroomCache/PrepareDenseScene/61c4f02547ec2c49ec920d480c2c166a0baa7318/mvs.ini"
 * maxInputPoints = 50000000
 * maxPoints = 5000000
 * maxPointsPerVoxel = 1000000
 * minAngleThreshold = 1
 * minStep = 2
 * output = "C:/Users/....../MeshroomCache/Meshing/79d6fdbd49f1152dbfd93532840c9392e179ff4e/mesh.obj"
 * partitioning =  Unknown Type "enum EPartitioningMode"
 * pixSizeMarginFinalCoef = 4
 * pixSizeMarginInitCoef = 2
 * refineFuse = 1
 * repartition =  Unknown Type "enum ERepartitionMode"
 * simFactor = 15
 * simGaussianSize = 10
 * simGaussianSizeInit = 10
 * verboseLevel = "info"
 * voteMarginFactor = 4

[15:37:09.160977][info] Found 1 image dimension(s):
[15:37:09.165975][info]  - [2736x2736]
[15:37:09.367901][info] Overall maximum dimension: [1368x1368]
[15:37:09.369897][warning] repartitionMode: 1
[15:37:09.370896][warning] partitioningMode: 2

**Comment by fabiencastan** [comment]
Could you try to reduce the value of `maxPoints` on the Meshing node to avoid using too much RAM & SWAP?

**Comment by shrobon** [comment]
@fabiencastan : 
Could you please assist as to how I can change maxPoints on the Meshing node using command line ?

**Comment by fabiencastan** [comment]
When you ask for command line, I suppose that you refer to https://github.com/alicevision/meshroom/blob/develop/bin/meshroom_photogrammetry.
The `maxPoints` parameter is not exposed, but you can easily add it. Look at the `--scale` parameter to see how to add a new one.

**Comment by fundamental4** [comment]
With the same pictures (61), I have a "Meshing" crash first, and secondly at "Depth Map Filter". 
I see your recommendations about reducing maxPoints, but:
- it is 5000000 now, how much do you recommend to be reduced? 
- what is maxPoints for? will the resolution be reduced?

**Comment by fundamental4** [comment]
16 gb RAM

De: Fabien Castan <notifications@github.com>
Enviado el: lunes, 10 de diciembre de 2018 12:28
Para: alicevision/meshroom <meshroom@noreply.github.com>
CC: fundamental4 <ruben.sofe@hotmail.com>; Mention <mention@noreply.github.com>
Asunto: Re: [alicevision/meshroom] Stuck at meshing (#243)


@fundamental4<https://github.com/fundamental4> yes, 5M sounds good but it depends on how much RAM is available on your computer.
maxPoint is the maximum number of points used to do the tetrahedralization.
Yes, reducing maxPoints will reduce the final number of vertices by fusing more points together. So yes, you can consider that as a lower resolution. Lower resolution in output also mean better robustness.

**Comment by daboy1** [comment]
Thank you for your work guy. This issue seems to be solved for the majority of the people. It seems that I have the same problem but Im unable to implement the necessary changes to the program. Could you give an absolute novice any hints on how to reduce the maxpoints? Any help greatly appreciated.

**Comment by natowi** [comment]
@daboy1 
![maxpoints](https://user-images.githubusercontent.com/8138537/60525855-7b399980-9cef-11e9-8355-4a870f3eef7a.JPG)

**Comment by natowi** [comment]
@sarveshkoyande the comment is a few years old and probably outdated. 
https://github.com/alicevision/meshroom/blob/develop/bin/meshroom_batch

It was suggested to add an argument parser like this (not tested):

```
parser.add_argument('--maxPoints', type=int, default=5000000,
                    help='Overwrite maxPoints'
                         'By default (5000000): use pipeline default value. '
                         'Valid range 100000-10000000, step 1000')
```

---
But paramOverrides should do the trick, too:

--paramOverrides [NODETYPE:param=value NODEINSTANCE.param=value [NODETYPE:param=value NODEINSTANCE.param=value ...]]
                        Override specific parameters directly from the command
                        line (by node type or by node names).

**Comment by sarveshkoyande** [comment]
Hi can you please clarify a bit further.
( Paramoverrides is available with v2020.1.0)

Let's say, How do I change feature extraction describer preset to ultra?

I tried --paramOverrides FeatureExtraction.describerPreset = ultra 

Output is invalid param override.

**Comment by JonasMoros** [comment]
Mushroom does not even utilize my 128gb of DDR5 and it still fails on meshing. Even when I reduce max points.
